### DISCUSSION TITLE
Welcome to ViennaPS Discussions!

### DISCUSSION THREAD

**Discussion by tobre1** [ViennaTools/ViennaPS#84] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

**Comment by sirsdroes** [comment]
Thank you creating the ViennaPS simulator.  I am interested in using the Python version.  I would like to use Python but it seems that getting the package installed is not working for me.  


Should I just save myself trouble and learn to run this in C++?  

Or am I missing something fundamental like a user key I would need to purchase. ? 


I suspect I am simply missing an installed package in my environment or some basic configuration issue.

I am able to install ViennaLS but cannot get the ViennaPS to install.  

 I am encountering errors when I try to install using the following approach:

_```
git clone https://github.com/ViennaTools/ViennaPS.git
cd ViennaPS

pip install .
```_

**I receive the error that is noted elsewhere in the documentation:**

Building wheels for collected packages: ViennaPS
  Building wheel for ViennaPS (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for ViennaPS (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [19 lines of output]
      *** scikit-build-core 0.11.1 using CMake 4.0.0 (wheel)
      *** Configuring CMake...
      loading initial cache file build\CMakeInit.txt
      -- Building for: NMake Makefiles
      CMake Error at CMakeLists.txt:2 (project):
        Running

         'nmake' '-?'

        failed with:

         no such file or directory


      CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
      CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for ViennaPS
Failed to build ViennaPS
ERROR: Failed to build installable wheels for some pyproject.toml based projects (ViennaPS)

**Reply by tobre1** [reply]
Thanks for your interest in ViennaPS, and sorry you're running into trouble with the Python version.

From the error log, it looks like you're missing a C++ compiler and CMake is unable to proceed. The line:

```
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

usually means no compiler is available in your environment. Also, `nmake` is missing, which suggests you're on Windows and don't have the necessary build tools installed.

### Quick suggestions:

1. **Use the pre-built package**  
   If you don’t need to modify the source, it’s much easier to install ViennaPS from PyPI:
   ```
   pip install ViennaPS
   ```
   No need to build locally in most cases.

2. **If you *do* want to build locally**  
   Make sure you have a C++ compiler and CMake installed. On Windows, you can install the [Visual Studio Build Tools](https://visualstudio.microsoft.com/visual-cpp-build-tools/), and make sure to include the "C++ CMake tools for Windows" during installation.

3. **You don’t need any license key or purchase**  
   ViennaPS is fully open source and free to use. No hidden keys or licenses required.

If you let me know your OS and Python environment (e.g., Anaconda, venv, etc.), I can give more targeted advice. But in general: try the `pip install ViennaPS` first, and only build from source if you have a specific reason.

Let me know how it goes.

**Comment by sirsdroes** [comment]
Hello Tobias,

 

First- my sincere thanks for the rapid and targeted response.  

 

I do not need to modify the source code so I will stick with the PyPI installation method.

 

As you suggest, I have tried the PyPI method.  I used 

 

1.  pip install ViennaPS

 

That appeared to work successfully for the base environment.    

 

The results of the installation in the base environment is shown below:

 

(base) C:\Users\sdroe>pip install ViennaPS

Collecting ViennaPS

  Using cached viennaps-3.4.0-cp312-cp312-win_amd64.whl.metadata (13 kB)

Collecting ViennaLS>=4.3.0 (from ViennaPS)

  Using cached viennals-4.3.0-cp312-cp312-win_amd64.whl.metadata (8.1 kB)

Using cached viennaps-3.4.0-cp312-cp312-win_amd64.whl (38.1 MB)

Using cached viennals-4.3.0-cp312-cp312-win_amd64.whl (12.3 MB)

Installing collected packages: ViennaLS, ViennaPS

Successfully installed ViennaLS-4.3.0 ViennaPS-3.4.0

 

I then tried to import the packages.  The ViennaLS seems to import without an issue.  The following command runs without an error.

 

import viennals2d as vls

 

 

However, there seems to be an issue with importing the ViennaPS package.  When I try to import ViennaPS2d here is what occurs:

 

import viennaps2d as vps

 

---------------------------------------------------------------------------

ImportError                               Traceback (most recent call last)

Cell In[4], line 1

----> 1 import viennaps2d as vps

 

File ~\anaconda3\Lib\site-packages\viennaps2d\__init__.py:33

     31 from viennals2d import IntegrationSchemeEnum as IntegrationScheme

     32 from viennals2d import BoundaryConditionEnum as BoundaryType

---> 33 from .viennaps2d import *

     35 # Config file reader helper function

     36 def ReadConfigFile(fileName: str):

 

ImportError: DLL load failed while importing viennaps2d: The specified module could not be found.

 

 

I suspect something in my PATH but I am not sure where to look.

 

For reference I am using Anaconda on a Windows 11 Home system.  

 

Here is the conda info output:

 


(base) C:\Users\sdroe>conda info

			

active environment

:

base


active env location

:

C:\Users\sdroe\anaconda3


shell level

:

1


user config file

:

C:\Users\sdroe\.condarc


populated config files

:

C:\Users\sdroe\anaconda3\.condarc

		
C:\Users\sdroe\.condarc


conda version

:

24.11.3


conda-build version

:

24.9.0


python version

:

3.12.7.final.0


solver

:

libmamba (default)


virtual packages

:

__archspec=1=x86_64_v3

		
__conda=24.11.3=0

		
__win=0=0


base environment

:

C:\Users\sdroe\anaconda3  (writable)


conda av data dir

:

C:\Users\sdroe\anaconda3\etc\conda


conda av metadata url

:

None


channel URLs

:

https://repo.anaconda.com/pkgs/main/win-64

		
https://repo.anaconda.com/pkgs/main/noarch

		
https://repo.anaconda.com/pkgs/r/win-64

		
https://repo.anaconda.com/pkgs/r/noarch

		
https://repo.anaconda.com/pkgs/msys2/win-64

		
https://repo.anaconda.com/pkgs/msys2/noarch


package cache

:

C:\Users\sdroe\anaconda3\pkgs

		
C:\Users\sdroe\.conda\pkgs

		
C:\Users\sdroe\AppData\Local\conda\conda\pkgs


envs directories

:

C:\Users\sdroe\anaconda3\envs

		
C:\Users\sdroe\.conda\envs

		
C:\Users\sdroe\AppData\Local\conda\conda\envs


platform

:

win-64


user-agent

:

conda/24.11.3 requests/2.32.3 CPython/3.12.7 Windows/11 Windows/10.0.26100 solver/libmamba conda-libmamba-solver/24.9.0 libmambapy/1.5.8 aau/0.4.4 c/dhmNjWgxAp2xab1Xi5g6YQ s/iOWSaV23S9LLBwgODUW8fQ e/D0QApFcWKrjhCiniBUd1Og


administrator

:

FALSE


netrc file

:

None


offline mode

:

FALSE

 

 

Thanks again for the help.  

 

Steven Droes

**Reply by tobre1** [reply]
Hi Steven,

Thanks for the detailed report.

I can't reproduce this error on my side. Unfortunately, I don't have much experience using Anaconda on Windows, so it's hard for me to say exactly what's going wrong.

Here are a few things you could try:

- Check if the **Microsoft Visual C++ Redistributable for Visual Studio 2022 (x64)** is installed. Missing runtime libraries are a common cause of DLL load errors. You can download it from [this link](https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist) or install it through the Visual Studio Installer.

- Try a clean install in a new conda environment:
  ```bash
  conda create -n viennaps-test python=3.10
  conda activate viennaps-test
  pip install ViennaPS
  python -c "import viennaps2d"
  ```

- If that doesn't help, consider using [**WSL** (Windows Subsystem for Linux)](https://learn.microsoft.com/en-us/windows/wsl/install). ViennaPS should work there without issues.

Let me know if any of these help.

**Comment by sirsdroes** [comment]
Thanks Tobias-

 

I will give those suggestions a try and let you know what happens.  

 

Steve

**Comment by sirsdroes** [comment]
Hello Tobias, 

 

I shifted to   <https://learn.microsoft.com/en-us/windows/wsl/install> WSL (Windows Subsystem for Linux)

 

I was able to install and import ViennaPS.  I ran two of the example files – Atomic Layer Deposition and Wet Cantilever Etching.  Both ran without errors.

 

Thanks for the suggestion to shift to Linux.  

 

For now I will spend some time working through the examples to understand the theory and implementation of the code.  

 

Steve Droes

 

 

 

From: ***@***.*** ***@***.***> 
Sent: Thursday, April 10, 2025 9:14 AM
To: 'ViennaTools/ViennaPS' ***@***.***>; 'ViennaTools/ViennaPS' ***@***.***>
Cc: 'Comment' ***@***.***>
Subject: RE: [ViennaTools/ViennaPS] Welcome to ViennaPS Discussions! (Discussion #84)

 

Thanks Tobias-

 

I will give those suggestions a try and let you know what happens.  

 

Steve
